resource-managers/yarn/src/main/resources/META-INF/services/org.apache.spark.deploy.security.ServiceCredentialProvider

```diff
@@ -0,0 +1,3 @@
+org.apache.spark.deploy.yarn.security.YARNHadoopFSCredentialProvider
```

Comment:
> but I guess I could do that through parameterization

That would probably be better. You can see if using SparkHadoopUtil / YarnSparkHadoopUtil to handle the differences can help too.